web/ts/TUMLiveVjs.ts

```diff
@@ -343,10 +343,62 @@ export class StartInOverlay extends Component {
 
 export function jumpTo(hours: number, minutes: number, seconds: number) {
     videojs("my-video").ready(() => {
-        player.currentTime(hours * 60 * 60 + minutes * 60 + seconds);
+        player.currentTime(toSeconds(hours, minutes, seconds));
     });
 }
 
+export class VideoSections {
+    readonly streamID: number;
+
+    list: Section[];
+    currentHighlightIndex: number;
+
+    constructor(streamID) {
+        this.streamID = streamID;
+        this.list = [];
+        this.currentHighlightIndex = 0;
+    }
+
+    async fetch() {
+        await fetch(`/api/stream/${this.streamID}/sections`)
+            .then((res: Response) => {
+                if (!res.ok) {
+                    throw new Error("Could not fetch sections");
+                }
+                return res.json();
+            })
+            .then((sections) => {
+                this.list = sections;
+                attachCurrentTimeEvent(this);
+            })
+            .catch((err) => {
+                console.log(err);
+                this.list = [];
+                this.currentHighlightIndex = 0;
+            });
+    }
+}
+
+function attachCurrentTimeEvent(videoSection: VideoSections) {
+    player.ready(() => {
+        player.on("timeupdate", function () {
+            const currentTime = this.currentTime();
+            videoSection.currentHighlightIndex = videoSection.list.findIndex((section, i, list) => {
```

Comment:
Right now the player needs some time to become ready and load the progress (~1s for me), which causes the video section highlighting to jump when the progress is beyond the first section.
It would be cool if we can directly set the `currentHighlightIndex` according to the users progress, but that might be tricky do since we would probably need the player for the duration.

Maybe we can init `currentHighlightIndex` with something like `-1` (or make it nullable) and don't show the highlighting in this case in the beginning. When the user presses play we could then set `currentHighlightIndex` to a valid index and display it the highlighting.